pkg/component/runtime/state.go

```diff
@@ -189,6 +200,14 @@ func (s *ComponentState) syncExpected(comp *component.Component) bool {
 		s.expectedFeaturesIdx = 1
 	}
 
+	if !gproto.Equal(s.expectedComponent, comp.Component) {
+		changed = true
+		s.expectedComponentIdx++
+		s.expectedComponent = comp.Component
+	} else {
+		s.expectedComponentIdx = 1
```

Comment:
To be honest, I was just following the `features` field. So, it's just a copy-paste. If this is wrong, it's wrong in both.

Reply:
Yeah I question if this is correct as well. Why would it be reset to 1 when it is equal?

Reply:
@cmacknz what your referring to is a unit state change with setting a specific trigger, the component-level config is not supposed to be there. Do I need to put the index check elsewhere?

Reply:
I tend to agree that the reset to 1 doesn't look right here.  I'm looking through the PR where it was introduced for features to see if we had a good reason to do it, or if it's a mistake: https://github.com/elastic/elastic-agent/pull/2218.

Reply:
So, should I just delete the `else` block here? Do I understand it right?

Reply:
> @cmacknz what your referring to is a unit state change with setting a specific trigger, the component-level config is not supposed to be there. Do I need to put the index check elsewhere?

No it is fine as is for this change, once someone needs the component change to be detected outside of elastic-agent-client itself like features this will need to be added. Probably when the APM configuration is exposed.

Reply:
It's as @ycombinator said. I was looking with Paolo when he found it and I agreed it looked odd.

Reply:
Addressed in https://github.com/elastic/elastic-agent/pull/3179/commits/c7f0a62a1292b31276554e473b0b89dd8b242afd